x/cork/keeper/msg_server.go

```diff
@@ -62,3 +62,38 @@ func (k Keeper) SubmitCork(c context.Context, msg *types.MsgSubmitCorkRequest) (
 
 	return &types.MsgSubmitCorkResponse{}, nil
 }
+
+// ScheduleCork implements types.MsgServer
+func (k Keeper) ScheduleCork(c context.Context, msg *types.MsgScheduleCorkRequest) (*types.MsgScheduleCorkResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	if !k.HasCellarID(ctx, common.HexToAddress(msg.Cork.TargetContractAddress)) {
+		return nil, types.ErrUnmanagedCellarAddress
+	}
+
+	signer := msg.MustGetSigner()
+	validatorAddr, err := k.signerToValAddr(ctx, signer)
+	if err != nil {
+		return nil, err
+	}
+
+	k.SetScheduledCork(ctx, validatorAddr, msg.BlockHeight, *msg.Cork)
+	k.AddScheduledBlockHeight(ctx, msg.BlockHeight)
+
+	ctx.EventManager().EmitEvents(
+		sdk.Events{
+			sdk.NewEvent(
+				sdk.EventTypeMessage,
+				sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			),
+			sdk.NewEvent(
+				types.EventTypeCork,
```

Comment:
Would be good to have the block height as one of the fields in the emitted event.

Reply:
Added